src/domain/core/src/services/export_service.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::fmt::{Display, Formatter};
+use std::path::Path;
+
+use datafusion::dataframe::DataFrame;
+use internal_error::{BoxedError, InternalError};
+use thiserror::Error;
+
+use crate::*;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Service
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait ExportService: Send + Sync {
+    async fn export_to_fs(
+        &self,
+        df: DataFrame,
+        path: &Path,
+        format: ExportFormat,
+        partition_row_count: Option<usize>,
+    ) -> Result<u64, ExportError>;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub enum ExportFormat {
+    Parquet,
+    Csv,
+    NdJson,
+}
+
+impl Display for ExportFormat {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let str_val = match *self {
+            ExportFormat::Parquet => "parquet",
+            ExportFormat::Csv => "csv",
+            ExportFormat::NdJson => "ndjson",
+        };
+        f.write_str(str_val)
+    }
+}
```

Comment:
There is `strum` crate we use in many places, maybe it can help to avoid this boilerplate?